include/boost/geometry/algorithms/detail/buffer/get_piece_turns.hpp

```diff
@@ -149,44 +149,6 @@ class piece_turn_visitor
         return ! m_rings[piece1.first_seg_id.multi_index].has_concave;
     }
 
-
-    template <std::size_t Dimension, typename Iterator, typename Box>
-    inline void move_begin_iterator(Iterator& it_begin, Iterator it_beyond,
-                                    signed_size_type& index, int dir,
-                                    Box const& this_bounding_box,
-                                    Box const& other_bounding_box)
-    {
-        for(; it_begin != it_beyond
-                && it_begin + 1 != it_beyond
-                && detail::section::preceding<Dimension>(dir, *(it_begin + 1),
-                                                         this_bounding_box,
-                                                         other_bounding_box,
-                                                         m_robust_policy);
-            ++it_begin, index++)
-        {}
-    }
-
-    template <std::size_t Dimension, typename Iterator, typename Box>
-    inline void move_end_iterator(Iterator it_begin, Iterator& it_beyond,
-                                  int dir, Box const& this_bounding_box,
-                                  Box const& other_bounding_box)
-    {
-        while (it_beyond != it_begin
-            && it_beyond - 1 != it_begin
-            && it_beyond - 2 != it_begin)
-        {
-            if (detail::section::exceeding<Dimension>(dir, *(it_beyond - 2),
-                        this_bounding_box, other_bounding_box, m_robust_policy))
-            {
-                --it_beyond;
-            }
-            else
-            {
-                return;
-            }
-        }
-    }
-
```

Comment:
@awulkiew : I don't remember why this was there, indeed. It is around 7 years old. But I disabled it, and it (now) runs all tests without problems, so I removed it. I have to come back to it, because a similar thing is also in the non-buffered turns.

This most probably fixes the type conflict for you. I can compile with Boost MultiPrecision, and it runs but still gives many errors. So I will continue later with this. The Union (for example) runs fine with Boost MP, so indeed it has to be something with buffer.

Reply:
I think in general it could be very useful if we add minimal tests for all algorithms with some other type than `double` or `float`. Boost MP is a good choice I think.

Reply:
Yes, I think we can clean up old code (ttmath, GMP and some other types which were all not part of boost, but we used to verify that behaviour) and add MP there instead.
For union it works quite well (I think perfect, but some expectations need a small update). For buffer there is still a problem.

Reply:
It would make the testing more complete for sure. Though I had a different scenario in mind when I wrote the comment for the commit. In my branch I use MP types instead of robust/rescaled/integral coordinates to avoid overflow. So the original coordinate type is e.g. `double`. Still in both cases less-comparison would fail to compile. The original MP coordinate type would also cause error in rescaling function because AFAIR there is no implicit conversion from MP to fundamental types.

Reply:
@barendgehrels It looks like some kind of optimization searching for subranges in both rings corresponding to bounding boxes of sections in order to ignore the rest of the rings. So my guess is that you could notice a difference in performance in some cases. Still this was probably a premature optimization so I'm ok with removing it.

Reply:
I'll test more thoroughly coming week.